### PR TITLE
Fixes #1800. TextView now uses the same colors as TextField.

### LINKED ISSUE #1800: TextView: Default background color has usability issues
In Wizards Scenario, I have what is essentially a modal dialog box with a TextView filling the client area:

image

Note how it's actually very hard to tell that this is a non-read-only/editible TextView. 

I think `TextView` should use the same colors as `TextField`:

image

### PR DESCRIPTION
**Pull request by BDisp**
Fixes #1800.

## Pull Request checklist:

- [x] I've named my PR in the form of "Fixes #issue. Terse description."
- [x] My code follows the [style guidelines of Terminal.Gui](https://github.com/gui-cs/Terminal.Gui/blob/develop/.editorconfig) - if you use Visual Studio, hit `CTRL-K-D` to automatically reformat your files before committing.
- [x] My code follows the [Terminal.Gui library design guidelines](https://github.com/gui-cs/Terminal.Gui/blob/develop/CONTRIBUTING.md)
- [x] I ran `dotnet test` before commit
- [x] I have made corresponding changes to the API documentation (using `///` style comments)
- [x] My changes generate no new warnings
- [x] I have checked my code and corrected any poor grammar or misspellings
- [x] I conducted basic QA to assure all features are working

### CODE REVIEW

**Comment by BDisp** [comment]
This commit https://github.com/gui-cs/Terminal.Gui/pull/1971/commits/3308ab4d1054a45f3a7479cadaf2e20ffb2cd0cc will be needed after the pr #1972 is merged.

File: Terminal.Gui/Views/TextView.cs
Comment (tig) [inline Terminal.Gui/Views/TextView.cs:1975]: Should this be named `SetReadOnlyColor` instead?

Maybe since they are all currently `protected` we use this PR to fix all these methods to use correct naming?

Comment (tig) [inline Terminal.Gui/Views/TextView.cs:1975]: Why does this take a `List<Rune> line` and `int idx` pram, if they are not used?

File: Terminal.Gui/Windows/Wizard.cs
Comment (tig) [inline Terminal.Gui/Windows/Wizard.cs:206]: Why is this change part of this PR?

File: UICatalog/Scenarios/Wizards.cs
Comment (tig) [inline UICatalog/Scenarios/Wizards.cs:220]: I think changes to Wizard should be in a separate PR.

**Review by tig — CHANGES_REQUESTED** [review]
Please see my comments.

File: Terminal.Gui/Views/TextView.cs
Comment (BDisp) [reply Terminal.Gui/Views/TextView.cs:1975]: I think @tznind only provide them for overridden them, see the `SyntaxHighlighting` scenario.

Comment (BDisp) [reply Terminal.Gui/Views/TextView.cs:1975]: I agree. I will change them.

File: Terminal.Gui/Windows/Wizard.cs
Comment (BDisp) [reply Terminal.Gui/Windows/Wizard.cs:206]: Because it only have the appropriate effect with this PR.

File: UICatalog/Scenarios/Wizards.cs
Comment (BDisp) [reply UICatalog/Scenarios/Wizards.cs:220]: Because it only have the appropriate effect with this PR. But if you insist I'll do a new PR with this PR merged. I'll wait for your feedback.

File: Terminal.Gui/Windows/Wizard.cs
Comment (BDisp) [reply Terminal.Gui/Windows/Wizard.cs:206]: Yes it does, because the Wizard use the `TextView` and you changed the `ColorScheme` to looks better with the old `TextView` `ColorScheme`. With this PR, the changes some of their colors will affect how the `Wizards` scenario will display with this PR.

File: Terminal.Gui/Views/TextView.cs
Comment (tznind) [reply Terminal.Gui/Views/TextView.cs]: So this method is declared on the `TextView` and is part of the render loop.  It is called for each index in the line.  It allows the user to alter the color set on the `Driver` based on arbitrary requirements (e.g. syntax highlighting).  

So for this reason it is not called `SetNormalColor` because it doesn't change the Normal color of the view.  It only changes the Color that is being used for that part of the current line as it is being rendered. 

I guess you could call it `SetDriverToNormalColor` though? 

It was implemented in this PR: https://github.com/gui-cs/Terminal.Gui/pull/1336

Comment (BDisp) [reply Terminal.Gui/Views/TextView.cs]: I think the changes @tig requested and which I already did is enough. The explanation for the `line` and `idx` parameters not being used but handled on overridden methods are clarified.

Comment (BDisp) [reply Terminal.Gui/Views/TextView.cs]: Done.

File: Terminal.Gui/Windows/Wizard.cs
Comment (tig) [reply Terminal.Gui/Windows/Wizard.cs:206]: Please answer this question: 

Did you make these changes to fix Wizard, or to fix a broader problem in TextView and TextField?

Comment (BDisp) [reply Terminal.Gui/Windows/Wizard.cs:206]: @tig I did to fix `Wizard` after fixing a broader problem in `TextView` and `TextField`.

Comment (tig) [reply Terminal.Gui/Windows/Wizard.cs:206]: > @tig I did to fix `Wizard` after fixing a broader problem in `TextView` and `TextField`.

I'm still not getting my thoughts to you clearly. My apologies.

I do not think this PR should include changes to Wizard. I think it should be two PRs, fixing two Issues:

1) TextView/TextField use different colors.
2) Wizard no longer needs to set specific colors, because 1) has been fixed.

Comment (BDisp) [reply Terminal.Gui/Windows/Wizard.cs:206]: > 2\. Wizard no longer needs to set specific colors, because 1) has been fixed.

Done #2012 teacher @tig :-)

**Review by tig — APPROVED** [review]

**Comment by arakis** [comment]
What is the "right" way to get the old colors / Behavior?

Old Behavior: Blue Background, Gray Text.
Now Gray Background, Darkgray Text.

**Comment by tznind** [comment]
> Old Behavior: Blue Background, Gray Text. Now Gray Background, Darkgray Text.
> 
> In my case, I have a border/panel around the view, it looks much more natural to have the background blue.

You can set the colors by creating a `ColorScheme` and assigning it to the `View`.  Make sure to only declare after you have called `Application.Init ();`

For example:
```csharp
using Terminal.Gui;
using Attribute = Terminal.Gui.Attribute;

Application.Init();
var win = new Window("Example App (Ctrl+Q to quit)");

ColorScheme myscheme = new ColorScheme
{
    // Color when NOT tabbed into
    Normal = new Attribute(
        Color.Red,Color.Black),

    // Selected text color when NOT tabbed into
    HotNormal = new Attribute(
        Color.Black,Color.Red),

    // Color when tabbed into
    Focus = new Attribute(
        Color.Red,Color.Black),
    
    // Selected text color when tabbed into
    HotFocus = new Attribute(
        Color.Black,Color.Red),

    // When disabled
    Disabled = new Attribute(Color.Gray,Color.Black),
    
};
win.Add(
    new TextView()
    {
        Width = Dim.Fill(),
        Height = Dim.Fill() -1,
        ColorScheme = myscheme,
        AllowsTab = false,
    }
);
win.Add(
    new TextField()
    {
        Width = 10,
        Y = Pos.AnchorEnd(1)
    }
);

Application.Run(win);
Application.Shutdown();

```
redtext

**Pull request merged**
